server/sources.go

```diff
@@ -142,10 +180,28 @@ func (s *Service) Sources(w http.ResponseWriter, r *http.Request) {
 		Sources: make([]sourceResponse, len(srcs)),
 	}
 
-	for i, src := range srcs {
-		res.Sources[i] = newSourceResponse(src)
+	wg := sync.WaitGroup{}
```

Comment:
I don't think we need wait groups here nor go routines.

I think you are protected with your timeouts in the LDAP config.

Reply:
the timeout addresses the if any of them timeout whereas the go routines and waitgroup here are to ensure if we have multiple sources we query for them in parallel.

Reply:
Yup, I understand, but, I worry it is adding complexity of concurrency when most sources are hard-coded values.

What do you think about adding this when there is a request to make loading sources faster?